openpype/hosts/fusion/plugins/publish/validate_saver_resolution.py

```diff
@@ -0,0 +1,96 @@
+)
+
+from openpype.hosts.fusion.api.action import SelectInvalidAction
+from openpype.hosts.fusion.api import get_current_comp
+
+
+class ValidateSaverResolution(
+    pyblish.api.InstancePlugin, OptionalPyblishPluginMixin
+):
+    """Validate that the saver input resolution matches the projects"""
+
+    order = pyblish.api.ValidatorOrder
+    label = "Validate Saver Resolution"
+    families = ["render"]
+    hosts = ["fusion"]
+    optional = True
+    actions = [SelectInvalidAction]
+
+    @classmethod
+    def get_invalid(cls, instance, wrong_resolution=None):
+        if wrong_resolution is None:
+            wrong_resolution = []
+
+        saver = instance[0]
+        firstFrame = saver.GetAttrs("TOOLNT_Region_Start")[1]
+        comp = get_current_comp()
+
+        # If the current saver hasn't bin rendered its input resolution
+        # hasn't bin saved. To combat this, add an expression in
+        # the comments field to read the resolution
+
+        # Create undo stack so we later can remove all changes
+        comp.StartUndo("Read resolution")
+
+        # Save old comment
+        oldComment = ""
+        hasExpression = False
+        if saver["Comments"][firstFrame] != "":
+            if saver["Comments"].GetExpression() is not None:
+                hasExpression = True
+                oldComment = saver["Comments"].GetExpression()
+                saver["Comments"].SetExpression(None)
+            else:
+                oldComment = saver["Comments"][firstFrame]
+                saver["Comments"][firstFrame] = ""
+
+        # Get input width
+        saver["Comments"].SetExpression("self.Input.OriginalWidth")
+        width = int(saver["Comments"][firstFrame])
+
+        # Get input height
+        saver["Comments"].SetExpression("self.Input.OriginalHeight")
+        height = int(saver["Comments"][firstFrame])
+
+        # Reset old comment
+        saver["Comments"].SetExpression(None)
+        if hasExpression:
+            saver["Comments"].SetExpression(oldComment)
+        else:
+            saver["Comments"][firstFrame] = oldComment
+
+        # False undo removes the undo-stack from the undo list
+        comp.EndUndo(False)
+
+        # Time to compare!
+        wrong_resolution.append("{}x{}".format(width, height))
+        entityData = instance.data["assetEntity"]["data"]
+        if entityData["resolutionWidth"] != width:
+            return [saver]
+        if entityData["resolutionHeight"] != height:
+            return [saver]
+
+        return []
+
+    def process(self, instance):
+        if not self.is_active(instance.data):
+            return
+
+        wrong_resolution = []
+        invalid = self.get_invalid(instance, wrong_resolution)
+        if invalid:
+            entityData = instance.data["assetEntity"]["data"]
+            raise PublishValidationError(
+                "The input's resolution does not match"
+                " the project's resolution of {}x{}.\n\n"
```

Comment:
It's not the project's resolution but the asset's resolution so please refactor project to asset where you mention it like that.

Reply:
Thanks for pointing it out!